firmware/ec/src/devices/at45db.c

```diff
@@ -0,0 +1,284 @@
+ **
+ **    ARGUMENTS       : spi device configuration, cmd buffer, register value,
+ **                      page offset, numOfBytes to be written, cmd write count.
+ **
+ **    RETURN TYPE     : Success or failure
+ **
+ *****************************************************************************/
+static ReturnStatus AT45DB_write_reg(AT45DB_Dev *dev,
+                                     void *cmdbuffer, /* cmd or opcode buffer */
+                                     uint8_t *regValue,
+                                     uint32_t pageOffset,
+                                     uint32_t NumOfbytes,
+                                     uint8_t writeCount)
+{
+    ReturnStatus status = RETURN_NOTOK;
+    SPI_Handle at45dbHandle = spi_get_handle(dev->cfg.dev.bus);
+    if (!at45dbHandle) {
+        LOGGER_ERROR("AT45DBFLASHMEMORY:ERROR:: Failed to get SPI Bus for at45db flash memory "
+                     "0x%x on bus 0x%x.\n", dev->cfg.dev.chip_select,
+                     dev->cfg.dev.bus);
+    } else {
+        status = spi_reg_write(at45dbHandle,
+                               dev->cfg.dev.chip_select,
+                               cmdbuffer,
+                               regValue,
+                               NumOfbytes,
+                               pageOffset,
+                               writeCount);
+    }
+    return status;
+}
+
+/*****************************************************************************
+ **    FUNCTION NAME   : at45db_readStatusRegister
+ **
+ **    DESCRIPTION     : Reads status of at45db device whether it is ready for
+ **
+ **                      r/w operation
+ **
+ **    ARGUMENTS       : spi device configuration
+ **
+ **    RETURN TYPE     : 8-bit status code
+ **
+ *****************************************************************************/
+uint8_t at45db_readStatusRegister(AT45DB_Dev *dev)
+{
+    uint8_t txBuffer = AT45DB_STATUS_OPCODE; /* opcode for ready status of AT45DB */;
+    uint8_t status;
+
+    AT45DB_read_reg(dev, &txBuffer, &status, NULL, AT45DB_STATUS_RD_BYTES, AT45DB_STATUS_OPCODE_WR_COUNT);
+
+    return (status);
+}
+
+/*****************************************************************************
+ **    FUNCTION NAME   : at45db_erasePage
+ **
+ **    DESCRIPTION     : Erases at45db memory page before writing data to it
+ **
+ **    ARGUMENTS       : spi device configuration, page number to be erased
+ **
+ **    RETURN TYPE     : Success or failure
+ **
+ *****************************************************************************/
+ReturnStatus at45db_erasePage(AT45DB_Dev *dev, uint32_t page)
+{
+    ReturnStatus status = RETURN_NOTOK;
+    uint8_t txBuffer[4];
+
+    waitForReady(dev);
+
+    txBuffer[0] = AT45DB_PAGE_ERASE_OPCODE; /* opcode to erase main memory page */
+    txBuffer[1] = (uint8_t)(page >> 7);  /* Page size is 15 bits 8 in tx1 and 7 in tx2 */
+    txBuffer[2] = (uint8_t)(page << 1);
+    txBuffer[3] = 0x00;
```

Comment:
There is no standard about this, but most common usage in our code base is lower case x, e.g. 0x00